core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -90,6 +90,8 @@ private static Map<String, String> persistedProperties(Map<String, String> rawPr
     persistedProperties.put(
         TableProperties.PARQUET_COMPRESSION,
         TableProperties.PARQUET_COMPRESSION_DEFAULT_SINCE_1_4_0);
+    persistedProperties.put(
+        TableProperties.DELETE_GRANULARITY, TableProperties.DELETE_GRANULARITY_DEFAULT_SINCE_1_8_0);
```

Comment:
This matches what we did for zstd in Parquet. I think it should be fine. An alternative is to change the default in our Spark conf classes, if we want to make this specific to Spark. In theory, it would allow us to change this value only for particular versions of Spark (if we don't want to cherry-pick this to older Spark versions).

```
public DeleteGranularity deleteGranularity() {
  String valueAsString =
      confParser
          .stringConf()
          .option(SparkWriteOptions.DELETE_GRANULARITY)
          .tableProperty(TableProperties.DELETE_GRANULARITY)
          .defaultValue(TableProperties.DELETE_GRANULARITY_DEFAULT) // this will have to change
          .parse();
  return DeleteGranularity.fromString(valueAsString);
}
```